server/incore-common/src/main/java/edu/illinois/ncsa/incore/common/auth/Authorizer.java

```diff
@@ -193,46 +195,37 @@ public boolean canUserWriteMember(String username, String memberId, List<Space>
     /**
      * This method determines if a user is an admin or not.
      *
-     * @param username string name of the user
+     * @param userGroups list of strings of groups user belongs to
      * @return boolean - true if the user is admin
      */
-    public boolean isUserAdmin(String username) {
-        try {
-            LdapClient ldapClient = getLdapClient();
-            Set<String> userGroups = ldapClient.getUserGroups(username);
-
-            return userGroups.contains(System.getenv("AUTH_LDAP_ADMINS"));
-        } catch (Exception e) {
-            logger.error(e);
-            throw new IncoreHTTPException(Response.Status.INTERNAL_SERVER_ERROR, "Internal server error.");
-        }
+    public boolean isUserAdmin(List<String> userGroups) {
+        return (userGroups.contains("incore_admin") || userGroups.contains("incore_ncsa"));
     }
 
     ////////////////////////////////////////////////////////
     // private methods
     /////////////////////////////////////////////////////////
 
-    private Set<PrivilegeLevel> getGroupSpecificPrivileges(String user, Privileges spec) {
+    private Set<PrivilegeLevel> getGroupSpecificPrivileges(String user, Privileges spec, List<String> userGroups) {
         if (spec == null) {
             return allowThisUser(user);
         }
         try {
-            if (isUserAdmin(user)) {
+            if (isUserAdmin(userGroups)) {
                 HashSet<PrivilegeLevel> admin = new HashSet<>();
                 admin.add(PrivilegeLevel.ADMIN);
                 admin.add(PrivilegeLevel.WRITE);
                 admin.add(PrivilegeLevel.READ);
                 return admin;
             }
 
-            Set<String> userGroups = ldapClient.getUserGroups(user);
             Set<PrivilegeLevel> privs = spec.groupPrivileges.keySet().stream()
                 .filter(userGroups::contains)
                 .map(key -> spec.groupPrivileges.get(key))
                 .collect(Collectors.toSet());
 
             //if the user is in the magic view-all group, give them read access
-            if (userGroups.contains(System.getenv("AUTH_LDAP_VIEW_ALL"))) {
+            if (userGroups.contains("incore_admin") || userGroups.contains("incore_ncsa")) {
```

Comment:
same here

Reply:
this `AUTH_LDAP_VIEW_ALL` in the past have given us so much trouble. Can we just use the existing `AUTH_LDAP_ADMINS` instead? @robkooper